lib/dry/web/settings.rb

```diff
@@ -0,0 +1,59 @@
+require "yaml"
+
+module Dry
+  module Web
+    class Settings
+      TypeError = Class.new(StandardError)
+
+      AnyType = Class.new do
+        def self.[](value)
+          value
+        end
+      end
```

Comment:
We should probably add `Dry::Types::Object` for such use-cases, could you add a TODO comment so that we won't forget to update it once this type is in place?